### PR TITLE
NAS-108441 / 12.0 /  Do not include system dataset into quota alerts

### PR DESCRIPTION
**Pull request by themylogin**
I am not sure we should have a new alert in 12.0. What if an average system has a core file there? (I had one for syslog-ng for example) We'll have a ton of tickets!

### CODE REVIEW

**Comment by themylogin** [comment]
@freenas/docs-team please write proper title and text for new alert, thank you.

File: src/middlewared/middlewared/alert/source/cores.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import os
+
+from middlewared.alert.base import AlertClass, AlertCategory, AlertLevel, Alert, ThreadedAlertSource
+
+
+class CoreFilesArePresentAlertClass(AlertClass):
+    category = AlertCategory.SYSTEM
+    level = AlertLevel.WARNING
+    title = "Core Files Are Present"
+    text = ("Core files are present in /var/db/system/cores: %s. Please, file a ticket at https://jira.ixsystems.com/ "
```
Comment (william-gr) [inline src/middlewared/middlewared/alert/source/cores.py:10]: Not sure if there is any usefulness in listing every file here.

**Review by william-gr — COMMENTED** [review]
I agree with you.

I wonder if we should compare the creation date of these files and only account for relatively newer ones... e.g. within last 5 days?

Also syslog-ng is known to segfault a LOT for unknown reasons, may make sense to blacklist that one on BSD.

Thoughts?

File: src/middlewared/middlewared/alert/source/cores.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import os
+
+from middlewared.alert.base import AlertClass, AlertCategory, AlertLevel, Alert, ThreadedAlertSource
+
+
+class CoreFilesArePresentAlertClass(AlertClass):
+    category = AlertCategory.SYSTEM
+    level = AlertLevel.WARNING
+    title = "Core Files Are Present"
```
Comment (Mrt134) [inline src/middlewared/middlewared/alert/source/cores.py:9]:
```suggestion
    title = "Core Files Found in System Database"
```

File: src/middlewared/middlewared/alert/source/cores.py (hunk 1 above)
Comment (themylogin) [reply src/middlewared/middlewared/alert/source/cores.py:10]: > I wonder if we should compare the creation date of these files and only account for relatively newer ones... e.g. within last 5 days?

I don't think that'll be right, files older than 5 days are still consuming disk space

File: src/middlewared/middlewared/alert/source/cores.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    text = ("Core files are present in /var/db/system/cores: %s. Please, file a ticket at https://jira.ixsystems.com/ "
+            "and then remove them.")
```
Comment (Mrt134) [inline src/middlewared/middlewared/alert/source/cores.py:11]:
```suggestion
    text = ("System core files were found in /var/db/system/cores: %s. Please create a ticket at https://jira.ixsystems.com/ "
            "and remove these files.")
```

**Review by Mrt134 — APPROVED** [review]
I put in a few suggestions for title and text

File: src/middlewared/middlewared/alert/source/cores.py (hunk 1 above)
Comment (william-gr) [reply src/middlewared/middlewared/alert/source/cores.py:10]: We could delete older files?
Not sure its useful to keep old files. i have seen reports back fro.2018

File: src/middlewared/middlewared/alert/source/cores.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    def check_sync(self):
+        cores = "/var/db/system/cores"
+        try:
+            listdir = os.listdir(cores)
+        except Exception:
+            return
+
+        has_core_file = False
+        for file in listdir:
+            path = os.path.join(cores, file)
```
Comment (william-gr) [inline src/middlewared/middlewared/alert/source/cores.py:26]: Wonder if we should make sure its a file and not a dir, and ending with .core?

**Comment by william-gr** [comment]
@anodos325 @yocalebo curious on yours opinion on this

File: src/middlewared/middlewared/alert/source/cores.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        except Exception:
+            return
+
+        has_core_file = False
+        for file in listdir:
+            if not file.endswith(".core"):
+                continue
+
+            path = os.path.join(cores, file)
+            if not os.path.isdir(path):
```
Comment (william-gr) [inline src/middlewared/middlewared/alert/source/cores.py:30]: you mean .isfile?

File: src/middlewared/middlewared/alert/source/cores.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+                continue
+
+            unlink = False
+            if IS_FREEBSD and file == "syslog-ng.code":
```
Comment (william-gr) [inline src/middlewared/middlewared/alert/source/cores.py:34]: .core

**Comment by yocalebo** [comment]
`syslog-ng` should definitely be blacklisted. How is this alert removed? Removing all the core files in that directory and then letting the period alert task run and detect there are none?

**Comment by themylogin** [comment]
@yocalebo yes, and period is 1 minute

**Review by yocalebo — APPROVED** [review]
I'm okay with this change, however, I predict we will be seeing a few more tickets because of this 😄

**Comment by william-gr** [comment]
Me too but I couldn't think of anything better.

The cores dataset was getting full of cores and we were getting alerts of zfs quota on that.
The alternative was ignoring cores dataset being full and unable to core dump anymore.

Open to new/other ideas

**Comment by yocalebo** [comment]
This approach is better than ignoring cores dataset and preventing core dumps for sure. I agree with the change as this will probably give us some insight into common applications that are core dumping....for which we should fix obviously.

**Review by william-gr — APPROVED** [review]

**Pull request merged**